From e2e655cdc588457ede0e1bec070d85d531ad6bdd Mon Sep 17 00:00:00 2001
From: Valentin David <valentin.david@codethink.co.uk>
Date: Tue, 8 Sep 2020 17:15:30 +0200
Subject: [PATCH 1/1] build/docs: fix generated XML syntax in
 tools/generate-docs-nm-settings-docs-gir.py

Class description may contains double quotes which is not valid as XML
attribute value.

[thaller@redhat.com: adjust original patch to reformat code with python black]

https://gitlab.freedesktop.org/NetworkManager/NetworkManager/-/merge_requests/620
(cherry picked from commit 54e25f23f53af889703dfc50d51a8afeeea8a439)
(cherry picked from commit 7d26773138dd4270c45e131761d4e6be58f64d12)
---
 tools/generate-docs-nm-settings-docs-gir.py | 8 ++++++--
 1 file changed, 6 insertions(+), 2 deletions(-)

diff --git a/tools/generate-docs-nm-settings-docs-gir.py b/tools/generate-docs-nm-settings-docs-gir.py
index ebda99560a70..317b32972a46 100755
--- a/tools/generate-docs-nm-settings-docs-gir.py
+++ b/tools/generate-docs-nm-settings-docs-gir.py
@@ -244,8 +244,12 @@ for settingxml in settings:
             "%s needs a gtk-doc block with one-line description" % setting.props.name
         )
     outfile.write(
-        '  <setting name="%s" description="%s" name_upper="%s" >\n'
-        % (setting.props.name, class_desc, get_setting_name_define(settingxml))
+        '  <setting name="%s" description=%s name_upper="%s" >\n'
+        % (
+            setting.props.name,
+            xml_quoteattr(class_desc),
+            get_setting_name_define(settingxml),
+        )
     )
 
     setting_properties = {
-- 
2.26.2

